arrow/src/ipc/reader.rs

```diff
@@ -54,14 +54,15 @@ fn read_buffer(buf: &ipc::Buffer, a_data: &[u8]) -> Buffer {
 ///     - cast the 64-bit array to the appropriate data type
 fn create_array(
     nodes: &[ipc::FieldNode],
-    data_type: &DataType,
+    field: &Field,
     data: &[u8],
     buffers: &[ipc::Buffer],
-    dictionaries: &[Option<ArrayRef>],
+    dictionaries: &HashMap<i64, ArrayRef>,
```

Comment:
```suggestion
    dictionaries_by_id: &HashMap<i64, ArrayRef>,
```

Maybe this would more consistent with the names used in the rest of this PR